src/dream.mli

```diff
@@ -835,7 +835,7 @@ val form : request -> (string * string) list form_result promise
 (** {2 Upload} *)
 
 type part = [
-  | `Files of (string * string) list
+  | `File of string * string
```

Comment:
With a big picture, we can rollback that and "merge" files with the same name into one and unique `Files`. I did this change because it's more 1-to-1 to what `Multipart_form` gives to us - it returns a list of all parts (even if name of them appear several times).

Reply:
> "merge" files with the same name into one and unique `Files`. I did this change because it's more 1-to-1 to what `Multipart_form` gives to us - it returns a list of all parts (even if name of them appear several times).

This reads as if it is about merging files with the same filename into one file. But yes, I mean merging files with the same *field* name into a list of files for that field.

`multipart/form-data` is ultimately an implementation detail from the point of view of a framework user. What they see is that they emitted a form like

```html
<form ...>
  <input type="text" name="foo">
  <input type="file" name="bar" multiple>
</form>
```

and in the easy-to-use buffering helper, it should be obvious how to retrieve the whole value for each one of those `<input>` fields, without any boilerplate that will have to be repeated everywhere one wants to do so. So I think the files should be merged into a list (as they already were).

I also imagine that if we checked, we would find that all user agents and clients send the file parts for `<input multiple>` contiguously anyway, so it's not an unreasonable representation of a real `multipart/form-data` transmission to represent these parts as a list, either. Of course, someone could probably use curl to send parts completely out of order, but there is no reason AFAIK why we should represent that directly in the easy-to-use API.